### PR TITLE
Server lifetime fixes for `gulp serve`

### LINKED ISSUE #10013: Reliably shutdown server.js when gulp crashes
Lets put in some signal trap or similar to make sure that `build-system/server.js` always shuts down when gulp crashes. It often keeps lingering and then blocks restarting gulp because it hangs on to the port.

### PR DESCRIPTION
**Pull request by rsimha-amp**
This PR does two things:

1. When `gulp serve` is run, if the parent `gulp` process crashes, the child process running the server will automatically exit.
2. If the port required by `gulp serve` is already in use (by another `gulp serve` instance, or by some other process), we print a clear message and exit, instead of throwing an unhandled `EADDRINUSE` exception.

Fixes #10013

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "file-reader": "1.1.1",
     "formidable": "1.0.17",
     "fs-extra": "0.27.0",
+    "is-running": "2.1.0",
```
Comment (erwinmombay) [inline package.json]: please also update the yarn lock file

**Comment by rsimha-amp** [comment]
@erwinmombay, I ran `yarn upgrade`, and it seems to have made numerous, seemingly unnecessary changes to `yarn.lock`. Is this desirable, or is there a better way to update the file?

**Comment by jridgewell** [comment]
You should just kill it:

```js
const stream =  gulp.src(process.cwd()).pipe(webserver({});
process.once('SIGINT',function(){
  stream.emit('kill');
});
```

**Comment by rsimha-amp** [comment]
@jridgewell, that doesn't work if the parent `gulp` process crashes. There's no `SIGINT` signal on which to act from inside the child process.

**Comment by jridgewell** [comment]
Aren't they the same process? Where do we spawn a node instance for just this file?

**Comment by rsimha-amp** [comment]
Not the same process. `nodemon` spawns a detached child process, within which `gulp-webserver` is invoked.

**Comment by jridgewell** [comment]
In that case:

```diff
#build-system/tasks/serve.js
+ process.once('SIGINT', function () {
+  process.kill(process.pid);
+ });

**Comment by rsimha-amp** [comment]
@jridgewell, that won't work either, if the parent `gulp` process crashes.

**Pull request merged**